src/shim/telemetryClient.ts

```diff
@@ -14,29 +14,41 @@ import { Logger } from "./logging";
 import { Util } from "./util";
 import { AutoCollectConsole } from "./autoCollection/console";
 import { AutoCollectExceptions, parseStack } from "./autoCollection/exceptions";
-import { ApplicationInsightsOptions } from "../types";
+import { ApplicationInsightsOptions, ExtendedMetricType } from "../types";
 import { InternalConfig } from "./configuration/internal";
-
+import { IConfig } from "../shim/types";
+import Config = require("./configuration/config");
+import { dispose, Configuration, _setupCalled } from "./shim-applicationinsights";
+import { HttpInstrumentationConfig } from "@opentelemetry/instrumentation-http";
+import bunyan = require("./autoCollection/diagnostic-channel/bunyan.sub");
+import console = require("./autoCollection/diagnostic-channel/console.sub");
+import winston = require("./autoCollection/diagnostic-channel/winston.sub");
+import ConfigHelper = require("./util/configHelper");
 
 /**
  * Application Insights telemetry client provides interface to track telemetry items, register telemetry initializers and
  * and manually trigger immediate sending (flushing)
  */
 export class TelemetryClient {
-    private readonly _internalConfig: InternalConfig;
+    private _internalConfig: InternalConfig;
     private _options: ApplicationInsightsOptions;
     private _client: AzureMonitorOpenTelemetryClient;
     private _console: AutoCollectConsole;
     private _exceptions: AutoCollectExceptions;
     private _idGenerator: IdGenerator;
     public context: Context;
     public commonProperties: { [key: string]: string }; // TODO: Add setter so Resources are updated
+    public config: IConfig;
 
     /**
      * Constructs a new client of the client
      * @param setupString the Connection String or Instrumentation Key to use (read from environment variable if not specified)
      */
     constructor(input?: string | ApplicationInsightsOptions) {
```

Comment:
Remove ApplicationInsightsOptions parameter from here and leave same structure as old Application Insights

Reply:
That will break some things like the functional tests and the configuration example in the distro's README. Are we planning to release the shim after some changes to the distro's user configuration?